src/Cello/test_ViewMap.cpp

```diff
@@ -0,0 +1,765 @@
+
+    // execute the move Constructor
+    ViewMap<int> dest_view_map(std::move(src_manager.get()));
+
+    assign_range_3Darray(dest_view_map.at("C"), ValRange<int>(0, 1));
+
+    // perform the checks:
+    common_checks_
+      (dest_view_map, src_keys,
+       {ValRange<int>(0,0), ValRange<int>(17,-1), ValRange<int>(0,1)},
+       src_manager.view_vec());
+  }
+
+
+  template<bool srcContiguous, bool destContiguous>
+  void test_assignment() {
+    Unit::instance()->set_func("ViewMap", "operator=(ViewMap&&)");
+
+    // setup the src (that will be assigned)
+    const std::vector<std::string> src_keys = {"A", "B", "C"};
+    const std::array<int,3> src_shape = {4, 6, 8};
+    ManagedViewMap<int, srcContiguous> src_manager(src_keys, src_shape);
+
+    // setup the dest (that will be overwritten by the assignment)
+    const std::vector<std::string> dest_init_keys = {"C", "D", "E"};
+    const std::array<int,3> dest_init_shape = {3, 2, 1};
+    ManagedViewMap<int, destContiguous> dest_manager(dest_init_keys,
+                                                    dest_init_shape);
+    ViewMap<int>& dest_view_map = dest_manager.get();
+
+    assign_range_3Darray(src_manager.get().at("B"), ValRange<int>(17, -1));
+
+    // perform the actual assignment
+    dest_view_map = std::move(src_manager.get());
+
+    assign_range_3Darray(dest_view_map.at("C"), ValRange<int>(0, 1));
+
+    // perform the checks:
+    common_checks_
+      (dest_view_map, src_keys,
+       {ValRange<int>(0,0), ValRange<int>(17,-1), ValRange<int>(0,1)},
+       src_manager.view_vec());
+  }
+
+
+  void run_tests()
+  {
+    test_constructor<true>();
+    test_constructor<false>();
+
+    test_assignment<true, true>();
+    test_assignment<true, false>();
+    test_assignment<false, true>();
+    test_assignment<false, false>();
+  }
+};
+
+//----------------------------------------------------------------------
+
+class ImplicitCastTests{
+  // test implicit casts from ViewMap<T> to ViewMap<const T>
+
+private:
+
+  void test_cast_helper_(ViewMap<const int> dest_view_map,
+                         ViewMap<int>& src_view_map,
+                         const std::vector<std::string>& src_keys) {
+
+    // perform a check related to map-methods
+    unit_assert(consistent_key_order_(src_keys, dest_view_map));
+
+    // now let's perform some checks to ensure that the shallow copy operations
+    // work as expected
+    assign_range_3Darray(src_view_map.at("A"), ValRange<int>(17, -1));
+
+    // getting the signatures on the template functions correct so that they
+    // can be used to compare the template functions is challenging! So we just
+    // do it manually!
+    bool all_equal = true;
+    for (const std::string& key: src_keys) {
+      CelloView<const int,3> dest_view = dest_view_map.at(key);
+      CelloView<const int,3> src_view = src_view_map.at(key);
+
+      const int mz = dest_view.shape(0);
+      const int my = src_view.shape(1);
```

Comment:
```suggestion
      const int my = dest_view.shape(1);
```
I guess this really doesn't make a difference but src_view is a little unexpected when reading this.

Reply:
Good point! fixed